home/code_setup.py

```diff
@@ -0,0 +1,242 @@
+        update_code_visibility(change.owner.full_label, change.new)
+        if show_active.value:
+            render_table_with_filters()
+
+    def on_search_change(_):
+        render_table_with_filters()
+
+    def update_code_visibility(full_label, is_hidden):
+        """Update the visibility of a `Code` node."""
+        code_row = next(filter(lambda row: row["Full Label"] == full_label, data))
+        try:
+            code = orm.load_code(full_label)
+            code.is_hidden = is_hidden
+            code_row["Hide"] = is_hidden
+        except NotExistent:
+            row_index = data.index(code_row)
+            data.pop(row_index)
+            render_table_with_filters()
+
+    def unhide_all_codes():
+        for code in orm.Code.collection.all():
+            full_label = f"{code.label}@{code.computer.label}"
+            update_code_visibility(full_label, False)
+
+    def generate_page_buttons(total_pages):
+        def create_page_button(page_num):
+            button = ipw.Button(
+                description=str(page_num),
+                button_style="primary" if page_num == current_page.value else "",
+                layout=ipw.Layout(width="40px", height="36px"),
+                page_number=page_num,
+            )
+            button.on_click(on_page_change)
+            return button
+
+        return [create_page_button(i) for i in range(1, total_pages + 1)]
+
+    def filter_data(data, show_active, query):
+        """Filter the data based on active status and search query."""
+        filtered = data[:]
+
+        if show_active:
+            filtered = [row for row in data if not row["Hide"]]
+
+        if query:
+            query = query.strip().lower()
+            filtered = [
+                row
+                for row in data
+                if query in row["Full Label"].lower()
+                or query in row["Executable Path"].lower()
+            ]
+
+        return filtered
+
+    def render_table_with_filters():
+        """Render the table with current filters applied."""
+        visible_data = filter_data(data, show_active.value, search_box.value)
+
+        rpg = CONFIG["rows_per_page"]
+        total_pages = (len(visible_data) + rpg - 1) // rpg
+        current_page.max = max(total_pages, 1)
+
+        if total_pages == 0:
+            current_page.value = 1
+        else:
+            current_page.value = min(current_page.value, total_pages)
+
+        with table_output:
+            table_output.clear_output(wait=True)
+            display(render_table(visible_data, current_page.value, on_checkbox_change))
+
+        pagination.children = generate_page_buttons(total_pages)
+
+    table_output = ipw.Output()
+
+    show_active = ipw.Checkbox(
+        value=False,
+        description="Show active codes only",
+    )
+    show_active.observe(
+        on_show_active,
+        "value",
+    )
+
+    show_all_button = ipw.Button(
+        description="Show All",
```

Comment:
This button is not that clear for the user. Good to add a description in the GUI.

Reply:
I actually plan to change it. There was some discussion with Carlo regarding what is the best option (show all, hide all, etc.). Still thinking about it.

Reply:
Sorry, I realize now you mean the other button that unhides any hidden code. Okay, let me at least add a tooltip.